presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java

```diff
@@ -97,7 +97,7 @@ public double overlapPercentWith(StatisticRange other)
         double lengthOfIntersect = min(high, other.high) - max(low, other.low);
         if (isInfinite(lengthOfIntersect)) {
             if (isFinite(this.distinctValues) && isFinite(other.distinctValues)) {
-                return min(this.distinctValues, other.distinctValues) / max(this.distinctValues, other.distinctValues);
+                return min(other.distinctValues / this.distinctValues, 1);
```

Comment:
It does not look right. Any comment why previous formula was wrong and why this one is better?